sciath/launcher.py

```diff
@@ -48,135 +48,268 @@ def formatMPILaunchCommand(mpiLaunch,ranks,corespernode):
         launch = launch.replace("<ranks_per_node>",str(corespernode))
         launch = launch.replace("<RANKSPERNODE>",str(corespernode))
         launch = launch.replace("<RANKS_PER_NODE>",str(corespernode))
-    return(launch)
-
-def generateLaunch_PBS(accountname,queuename,testname,mpiLaunch,executable,ranks,ranks_per_node,walltime,outfile):
-    if not ranks:
-        print("<generateLaunch_PBS>: Requires the number of MPI-ranks be specified")
-    if not walltime:
-        print("<generateLaunch_PBS>: Requires the walltime be specified")
-
-    filename = testname + '-sciath.pbs'
+    return launch.split()
+
+
+# Returns name lists for error-code file (one per job), stdout, stderr
+def _getLaunchStandardOutputFileNames(job):
+    jobnames = []
+    try:
+        jobnames = job.createJobOrdering()
+    except:
+        jobnames.append(job.name)
+
+    errorCodeName = "sciath.job-" +  job.name + ".errorcode"
+    stdoutName = []
+    stderrName = []
+
+    lc_count = len(jobnames)
+    for i in range(0,len(jobnames)):
+        jprefix = "".join(["sciath.depjob-",str(lc_count),'-',jobnames[i]])
+        if lc_count == 1: # we do something special for the last job in a sequence/DAG list
+            jprefix = "sciath.job-" + job.name
+        
+        stdoutName.append( jprefix + ".stdout" )
+        stderrName.append( jprefix + ".stderr" )
+
+        lc_count -= 1
+
+    return errorCodeName, stdoutName, stderrName
+
+def _removeFile(file2rm):
```

Comment:
Is it safe to turn off "safety mode" here? I am trying to add your new tests to the self-tests before proceeding, but they need to run without user input.

Reply:
(For now, working around this by doing e.g. `yes | ./runTests.py`)